frontend/src/pages/SignUpPage.js

```diff
@@ -0,0 +1,137 @@
+import styled from "styled-components/macro";
+import { useState } from "react";
+import useSignUp from "../hooks/useSignUp";
+import {NavLink} from "react-router-dom";
+
+export default function SignUpPage() {
+    const {checkUserName, signUp, availability, success} = useSignUp();
+    const [error, setError] = useState()
+    const [credentials, setCredentials] = useState({
+        username: "",
+        password: "",
+        passwordcheck: "",
+    });
+
+    const handleChange = (event) => {
+        setCredentials({...credentials, [event.target.name]: event.target.value});
+    };
+
+    const handleSubmit = (event) => {
+        event.preventDefault();
+        if (credentials.password === "" || credentials.passwordcheck === "" || credentials.username === "") {
+            setError("Please fill out all fields");
+            return;
+        }
+        if (credentials.password !== credentials.passwordcheck) {
+            setError("Passwords do not match");
+            return;
+        } else {
+            checkUserName(credentials.username);
```

Comment:
Careful, I think this does not work the way you intended to. If I understand correctly, availability is set as a result of a network request. Said network request is async, and the next conditional is probably evaluated with the wrong value. Maybe return a promise from checkUserName and evaluate the data in a then block instead of storing it in a state?